src/logic/addressBook/utils/index.ts

```diff
@@ -62,12 +62,7 @@ export const getEntryIndex = (
   addressBookEntry: Overwrite<AddressBookEntry, { name?: string }>,
 ): number =>
   state.findIndex(
-    ({ address, chainId }) => chainId === addressBookEntry.chainId && sameAddress(address, addressBookEntry.address),
+    ({ address, chainId }) =>
+      chainId.toString() === addressBookEntry.chainId.toString() &&
+      checksumAddress(address) === checksumAddress(addressBookEntry.address),
```

Comment:
Nice! Is this all that was needed?

Reply:
I suggest we still use `sameAddress` but enhance it by adding `toLowerCase()` for each arg.

Reply:
I removed the `sameAddress` call and didn't include the `.toLowerCase()` because the `web3.utils.toChecksumAddress(address)` will convert the address to a checksum address regardless of the upper or lowercase. -> https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html#tochecksumaddress

Reply:
I suggested to use `toLowerCase` in `sameAddress` for two reasons:

 * it would improve `sameAddress` in general, there are many places where it's used
 * `toLowerCase()` is faster than checksumming, and doesn't throw, but achieves the same comparison

Reply:
Valid point. I will implement it as you suggest.

Reply:
Our method `sameAddress` uses `sameString` which in its turn is a case insensitive comparison so it is already working as we want.

https://github.com/gnosis/safe-react/blob/9f0e5da5a5f757aa0b4253813bb61c6164947c8d/src/utils/strings.ts#L55-L61

So by reverting this change, we are only filtering the imported address book in the `reducer`.